### PR TITLE
PIC-1688: Process Common Platform Hearings

### PR DESCRIPTION
**Pull request by jhevans**
This is now working and will post new cases to court-case-service and will attempt the search *but*:
- We can't fetch existing cases without caseNo so it will overwrite anything that already exists
- We can't post back matches for the same reason

I've added code to skip these steps for now and will follow up with tickets if they don't already exist.

### CODE REVIEW

File: src/test/java/uk/gov/justice/probation/courtcasematcher/restclient/model/courtcaseservice/CCSExtendedCaseTest.java [hunk 1]
```diff
@@ -83,4 +84,22 @@ public void shouldMapFromCourtCaseAndKeepIdsIfPresent() {
         assertThat(actualFirstDefendant.getAwaitingPsr()).isEqualTo(firstDefendant.getAwaitingPsr());
         assertThat(actualFirstDefendant.getBreach()).isEqualTo(firstDefendant.getBreach());
     }
+
+    @Test
```
Comment (chris-faulkner-mojdt) [inline src/test/java/uk/gov/justice/probation/courtcasematcher/restclient/model/courtcaseservice/CCSExtendedCaseTest.java:88]: Probably not caused by your change but I noticed that the source field isn't exercised by any of these tests in here.

File: src/test/java/uk/gov/justice/probation/courtcasematcher/messaging/MessageParserTest.java [hunk 2]
```diff
@@ -58,14 +58,14 @@
 
         @Autowired
         @Qualifier("commonPlatformJsonParser")
-        public MessageParser<CommonPlatformHearing> messageParser;
+        public MessageParser<CPHearing> messageParser;
```
Comment (chris-faulkner-mojdt) [inline src/test/java/uk/gov/justice/probation/courtcasematcher/messaging/MessageParserTest.java:61]: Again probably not part of your change but HEARING_DATE and START_TIME constants in this class are unused

**Review by chris-faulkner-mojdt — APPROVED** [review]
LGTM

File: src/test/java/uk/gov/justice/probation/courtcasematcher/restclient/model/courtcaseservice/CCSExtendedCaseTest.java (hunk 1 above)
Comment (jhevans) [reply src/test/java/uk/gov/justice/probation/courtcasematcher/restclient/model/courtcaseservice/CCSExtendedCaseTest.java:88]: Good spot, I added in source afterwards so must have missed it in this test. Will add it in

**Review by chris-faulkner-mojdt — APPROVED** [review]
LGTM

**Pull request merged**